### PR TITLE
sslstrip: add page

### PR DESCRIPTION
**Pull request by jimmychang851129**
- [ ] The page(s) are in the correct platform directories: `common`, `linux`, `osx`, `windows`, `sunos`, `android`, etc.
- [ ] The page(s) have at most 8 examples.
- [ ] The page description(s) have links to documentation or a homepage.
- [ ] The page(s) follow the [content guidelines](/tldr-pages/tldr/blob/main/CONTRIBUTING.md#guidelines).
- [ ] The PR title conforms to the recommended [templates](/tldr-pages/tldr/blob/main/CONTRIBUTING.md#commit-message-and-pr-title).
- **Version of the command being documented (if known):**

### CODE REVIEW

File: pages/linux/sslstrip.md
Comment (fazlearefin) [inline pages/linux/sslstrip.md:3]:
```suggestion
> An MITM tool that implements Moxie Marlinspike's SSL stripping attacks.
```

File: pages/linux/sslstrip.md
Comment (fazlearefin) [inline pages/linux/sslstrip.md:4]:
```suggestion
> More information: <https://github.com/L1ghtn1ng/sslstrip>.
```

Kali's link is not the official link for this project.

File: pages/linux/sslstrip.md
Comment (fazlearefin) [inline pages/linux/sslstrip.md:6]:
```suggestion
- Listen on a port:
```

Listen on is more common in tech.

File: pages/linux/sslstrip.md
Comment (vitorhcl) [reply pages/linux/sslstrip.md:3]:
```suggestion
> Perform Moxie Marlinspike's SSL stripping attacks with this MITM tool.
```

File: pages/linux/sslstrip.md
Comment (fazlearefin) [reply pages/linux/sslstrip.md:4]: @jimmychang851129 Using kali webpage breaks how we source more information. 

See https://github.com/tldr-pages/tldr/blob/main/contributing-guides/style-guide.md#more-information-links for details.

> On the More information link line, we prefer linking to the author's provided documentation of the command line reference or the man page. When not available, use https://manned.org/ as the default fallback for all platforms (except osx and BSD platforms other than FreeBSD). Alternatively, you can link to the author's website or a tutorial page if the command doesn't have a documentation page.

Comment (vitorhcl) [reply pages/linux/sslstrip.md:4]: IMO we should use Kali's documentation, the priority is the most complete resource, and not if it is the official one or not, that's why we use man pages sometimes.

It's kind of an informal rule, we should indeed document it.

Comment (vitorhcl) [reply pages/linux/sslstrip.md:4]: Also, when there is no documentation in the README, the style guide makes it ver clear:

> Alternatively, you can link to the author's website or a tutorial page if the command doesn't have a documentation page.

IMO a README without CLI usage can't be considered a documentation in this case.

Comment (fazlearefin) [reply pages/linux/sslstrip.md:4]: The author's github page contains a README (agreed it is not comprehensive) mentioning the important prerequisites needed to run the command which is missing in the Kali's link.

@jimmychang851129 Are you able to confirm if you were able to run the tool without the prerequisites mentioned in https://github.com/L1ghtn1ng/sslstrip/blob/master/README.md ?

Comment (kbdharun) [reply pages/linux/sslstrip.md:4]:
```suggestion
> More information: <https://manned.org/sslstrip.1>.
```

Manned has the [same Documentation](https://manned.org/sslstrip.1) but for an older version with no change in commands, we can link to it or to the Kali page instead of the author's page for this special case (as the author's docs don't have any documentation for [this file](https://github.com/L1ghtn1ng/sslstrip/blob/master/sslstrip.py)).

IMO as @vitorhcl says, Kali docs might be a better reference for this page.

---

Maybe, I will open an issue/PR to the author's repo for requesting/adding docs to README (referencing this PR).

File: pages/linux/sslstrip.md
Comment (kbdharun) [inline pages/linux/sslstrip.md:3]:
```suggestion
> Perform Moxie Marlinspike's Secure Sockets Layer (SSL) stripping attacks with this Man-in-the-Middle (MITM) tool.
```
For technical terms, we generally add abbreviations too for more beginner-friendliness for those who might not be aware of the terms. Feel free to reword my suggestion.

File: pages/linux/sslstrip.md
Comment (kbdharun) [inline pages/linux/sslstrip.md:6]:
```suggestion
- Listen on a specific port:
```

File: pages/linux/sslstrip.md
Comment (kbdharun) [inline pages/linux/sslstrip.md:10]:
```suggestion
- Log all SSL traffic to and from the server:
```

File: pages/linux/sslstrip.md
Comment (kbdharun) [inline pages/linux/sslstrip.md:14]:
```suggestion
- Log all SSL and HTTP traffic to and from the server:
```

File: pages/linux/sslstrip.md
Comment (kbdharun) [inline pages/linux/sslstrip.md:18]:
```suggestion
- Specify the output filename to store the logs:
```
How about this?

File: pages/linux/sslstrip.md
Comment (kbdharun) [inline pages/linux/sslstrip.md:20]:
```suggestion
`sslstrip --listen={{target_port}} --write={{path/to/file}}`
```
For files, we generally use a path placeholder. Feel free to checkout <https://github.com/tldr-pages/tldr/blob/main/contributing-guides/style-guide.md#paths>.

**Review by kbdharun — COMMENTED** [review]
Thanks for your contribution. I have some suggestions for this page.

File: pages/linux/sslstrip.md
Comment (fazlearefin) [reply pages/linux/sslstrip.md:4]: In case it got overlooked, note that Kali page do not contain the **PREREQUISITES** for running the command.

Comment (jimmychang851129) [reply pages/linux/sslstrip.md:4]: @fazlearefin  
I agree with the **prerequisites part** .  sslstrip is merely for deciphering Internet packet, and the prerequisite is required because users should intercept the Internet packet before deciphering it.

The idea is like, users should have a sql server running to have sqlmap to work. 
In short, the prerequisite is a must but is not related to the tool iteself.
Perhaps, there should be some way to inform users the prerequisite through tldr.  Any suggested format on this?

Comment (jimmychang851129) [reply pages/linux/sslstrip.md:4]: IMO,
Since tldr is not a penetrating tools manuals, the prerequisite (i.e. steps to intercept Internet packets) is not necessary to be shown on tldr, but I will edit the description to at least mention the prerequisite part.

Thanks a lot for brining up this, mentioning the prerequisite will be beneficial for the users.

**Comment by vitorhcl** [comment]
@kbdharun had an excellent idea and opened a PR to add an usage section in sslstrip's README.

However, I don't know when (and even if) it will be seen by the author because the last merged PR was 3 months ago.

I support using the official documentation on more info and, until @kbdharun's PR is merged, keeping a help example.

File: pages/linux/sslstrip.md
Comment (vitorhcl) [inline pages/linux/sslstrip.md:3]:
```suggestion
> Perform Moxie Marlinspike's Secure Sockets Layer (SSL) stripping attacks. Perform an ARP spoofing before using this tool.
```
- Litle typo (ARPspoofing).
- We generally use the imperative mood in the 2nd singular person instead of referring to the user in the 3rd person or using the indicative mood im the 2nd person.

File: pages/linux/sslstrip.md
Comment (vitorhcl) [inline pages/linux/sslstrip.md:20]:
```suggestion
`sslstrip --listen={{target_port}} --write={{path/to/file}}`

- Display help:

`sslstrip --help`
```

**Review by vitorhcl — APPROVED** [review]
LGTM after my suggestions

**Review by kbdharun — APPROVED** [review]

File: pages/linux/sslstrip.md
Comment (CleanMachine1) [inline pages/linux/sslstrip.md:3]:
```suggestion
> Hijack HTTP and HTTPS traffic on a network
```

You will have to correct me if this is wrong, I am unfamiliar with it.

However this description fits the style of a description

Also with the ARP spoofing suggestion, it may be better suited on its own line.

such as 

```
> Run a ARP spoofing attack in conjunction.
```

File: pages/linux/sslstrip.md
Comment (CleanMachine1) [inline pages/linux/sslstrip.md:8]:
```suggestion
`sslstrip --listen={{8080}}`
```

A suggestion port maybe better suited here, especially for newbs.

File: pages/linux/sslstrip.md
Comment (CleanMachine1) [inline pages/linux/sslstrip.md:10]: Maybe we can keep all of the descriptions consistent, the only variation from the one above is the --ssl flag.

So maybe

```suggestion
> Listen to HTTPS trafic on a specific port:
```

And apply this sort of change throughout.


Also it isn't currently logging really, since its not being wrote anywhere (from what I gather)

File: pages/linux/sslstrip.md
Comment (jimmychang851129) [reply pages/linux/sslstrip.md:3]: Thanks for the suggestion.
sslstrip does not hijack http/https traffic,  it basically decipher https traffic by downgrading the encryption protocol to the insecure one, so I will stick to the original description.

For the second suggestion, I would say we need to run ARP spoofing first to hijack traffic first. "In conjunction" may be a bit misleading.

File: pages/linux/sslstrip.md
Comment (jimmychang851129) [reply pages/linux/sslstrip.md:10]: The description is aligned with the kali document. IMO, it is better to use the document description rather than create our own description.

If the --write is not specified, the sslstrip will log it to a default file: sslstrip.log.

File: pages/linux/sslstrip.md
Comment (jimmychang851129) [reply pages/linux/sslstrip.md:3]: IMO, I prefer current description than the suggested one: "Hijack HTTP and HTTPS traffic". 

If the users want to learn more about this command, current description provides the key word to find related resources.

File: pages/linux/sslstrip.md
Comment (fazlearefin) [inline pages/linux/sslstrip.md:11]: `POST` is specific to HTTP/HTTPS. Maybe reword this?

Comment (jimmychang851129) [reply pages/linux/sslstrip.md:11]: You're right, thanks a lot for the comment. Will do.

**Review by kbdharun — APPROVED** [review]

**Pull request merged**
